### PR TITLE
migrate AuctionEndTime to double_map

### PR DESCRIPTION
**Pull request by ermalkaleci**
Closing #102

### CODE REVIEW

File: auction/src/lib.rs [hunk 1]
```diff
@@ -33,13 +28,11 @@ decl_event!(
...
 );
 
-type AuctionIdLinkedItem<T> = LinkedItem<<T as Trait>::AuctionId>;
-
 decl_storage! {
 	trait Store for Module<T: Trait> as Auction {
 		pub Auctions get(fn auctions): map hasher(twox_64_concat) T::AuctionId => Option<AuctionInfo<T::AccountId, T::Balance, T::BlockNumber>>;
 		pub AuctionsIndex get(fn auctions_index): T::AuctionId;
-		pub AuctionEndTime get(fn auction_end_time): map hasher(twox_64_concat) (T::BlockNumber, Option<T::AuctionId>) => Option<AuctionIdLinkedItem<T>>;
+		pub AuctionEndTime get(fn auction_end_time): double_map hasher(twox_64_concat) T::BlockNumber, hasher(twox_64_concat) T::AuctionId => Option<T::AuctionId>;
```
Comment (xlc) [inline auction/src/lib.rs]:
```suggestion
		pub AuctionEndTime get(fn auction_end_time): double_map hasher(twox_64_concat) T::BlockNumber, hasher(twox_64_concat) T::AuctionId => Option<()>;
```

It is now possible to iterate keys and values

File: auction/src/lib.rs [hunk 2]
```diff
@@ -104,12 +97,12 @@ decl_error! {
 
 impl<T: Trait> Module<T> {
 	fn _on_finalize(now: T::BlockNumber) {
-		let ended_auctions = <AuctionEndTimeList<T>>::take_all(&now);
-		ended_auctions.for_each(|auction_id| {
+		for auction_id in <AuctionEndTime<T>>::iter_prefix(&now) {
```
Comment (xlc) [inline auction/src/lib.rs]:
```suggestion
		for auction_id in <AuctionEndTime<T>>::drain(&now) {
```

File: auction/src/lib.rs (hunk 1 above)
Comment (ermalkaleci) [reply auction/src/lib.rs]: @xlc cool. didn't know that

Comment (xlc) [reply auction/src/lib.rs]: it is new feature is substrate alpha 5, which we just upgraded yesterday

**Pull request merged**